test/test.tools.js

```diff
@@ -5,6 +5,16 @@ let gmail = new Gmail();
 
 const testData = require("./testdata-parser.js");
 
+function testXhrEventParsing(jsonXhrData, eventName) {
+    const api = new Gmail();
+    const xhrData = JSON.parse(jsonXhrData);
+
+    const threads = api.tools.extract_from_graph(xhrData, api.check.data.is_thread);
+    const actionType = api.tools.check_event_type(threads[0]);
+
+    assert.equal(eventName, actionType);
+};
+
```

Comment:
This is a good test-helper. This looks good to me.

Reply:
this is your func)))